util.js

```diff
@@ -78,19 +45,47 @@ function fetch(url, options) {
  * @return {Promise} The promise to get a window object once the spec has
  *   been loaded with jsdom.
  */
-function loadSpecificationFromHtml(spec, counter) {
+async function loadSpecificationFromHtml(spec, counter) {
     let url = spec.url || 'about:blank';
     let responseUrl = spec.responseUrl || url;
     let html = spec.html || '';
     counter = counter || 0;
 
-    return new Promise((resolve, reject) => {
+    let promise = new Promise((resolve, reject) => {
         // Drop Byte-Order-Mark character if needed, it bugs JSDOM
         if (html.charCodeAt(0) === 0xFEFF) {
             html = html.substring(1);
         }
-        const {window} = new JSDOM(html, {url: responseUrl});
-        const doc = window.document;
+        const {window} = new JSDOM(html, {
+            url: responseUrl,
+            resources: 'usable',
+            runScripts: 'dangerously',
```

Comment:
thrilling!

Reply:
Yes, any interaction with dom is potentially dangerous, I've always said that :)